src/Microsoft.ML.StandardLearners/Standard/Simple/SimpleTrainers.cs

```diff
@@ -65,11 +63,35 @@ public RandomTrainer(IHostEnvironment env, Arguments args)
             Host.CheckValue(args, nameof(args));
         }
 
+        public BinaryPredictionTransformer<RandomPredictor> Fit(IDataView input)
+        {
+            var cachedTrain = Info.WantCaching ? new CacheDataView(Host, input, prefetch: null) : input;
+
+            RoleMappedData trainRoles = new RoleMappedData(cachedTrain);
+            var pred = Train(new TrainContext(trainRoles));
+            return new BinaryPredictionTransformer<RandomPredictor>(Host, pred, cachedTrain.Schema, trainRoles.Schema.Feature.Name);
```

Comment:
>trainRoles.Schema.Feature.Name [](start = 100, length = 30)

this will make features a required column, which we don't want

Reply:
Let's follow through and formally say that feature column is also optional in `PredictionTransformerBase`: right now it is not

---
In reply to: [216845418](https://github.com/dotnet/machinelearning/pull/876#discussion_r216845418) [](ancestors = 216845418)

Reply:
don't make it an optional parameter

---
In reply to: [217102915](https://github.com/dotnet/machinelearning/pull/876#discussion_r217102915) [](ancestors = 217102915,216845418)